src/NHSD.BuyingCatalogue.Organisations.Api/Services/MailKitEmailService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using MailKit;
+using NHSD.BuyingCatalogue.Organisations.Api.Settings;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    /// <summary>
+    /// A service for sending e-mails using MailKit.
+    /// </summary>
+    internal sealed class MailKitEmailService : IEmailService
+    {
+        private readonly IMailTransport _client;
+        private readonly SmtpSettings _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MailKitEmailService"/> class
+        /// using the provided <paramref name="client"/> and <paramref name="settings"/>.
+        /// </summary>
+        /// <param name="client">The mail transport to use to send e-mail.</param>
+        /// <param name="settings">The SMTP configuration.</param>
+        public MailKitEmailService(IMailTransport client, SmtpSettings settings)
+        {
+            _client = client;
```

Comment:
worth checking for null here as we do it everywhere else?

Reply:
I'm not sure we need to here.

The class is `internal` so (with the exception of tests) it should always be constructed by the DI container. If the `IMailTransport` dependency hasn't been wired up correctly then the DI container will throw an exception.

Reply:
I'm doing a reverse-ferret here because @sam-roberts-bjss asked a related question too.

While I'm in favour of failing fast I'm not always convinced by the value of precondition assertions in non-public code although I guess you could argue that's slightly contradictory... However, as both you and @sam-roberts-bjss have raised similar points I'm going to add them.

Possibly another one for code standards – do we want to do this everywhere? I guess nullable reference types might help here too (if we enabled that features).